packages/table/src/common/loadableContent.tsx

```diff
@@ -31,8 +31,13 @@ export class LoadableContent extends React.Component<ILoadableContentProps, {}>
 
     public constructor(props: ILoadableContentProps) {
         super(props);
-        const skeletonLength = props.variableLength ? 75 - Math.floor(Math.random() * 11) * 5 : 100;
-        this.style = { width: `${skeletonLength}%` };
+        this.calculateStyle(props.loading);
+    }
+
+    public componentWillReceiveProps(nextProps: ILoadableContentProps) {
+        if (!this.props.loading && nextProps.loading || this.props.variableLength !== nextProps.variableLength) {
+            this.calculateStyle(nextProps.variableLength);
```

Comment:
Since you use Math.random() every time in calculateStyle, won't this make the skeleton rectangle change size if the loading state switches back and forth?